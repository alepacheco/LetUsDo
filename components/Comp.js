function Home() {
  return <div>Imported</div>;
}

export default Home;