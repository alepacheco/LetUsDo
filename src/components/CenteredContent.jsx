import React from 'react';
import 'src/styles/components/CenteredContent.scss';

export const CenteredContent = ({ children, reverse }) => {
  return <div className={`centered-content ${reverse ? 'reverse' : ''}`}>{children}</div>;
};

export default CenteredContent;
