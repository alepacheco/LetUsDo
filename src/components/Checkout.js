import React from 'react';
import StripeCheckout from 'react-stripe-checkout';
import Button from 'react-bootstrap/Button';

export default class Checkout extends React.Component {
    constructor(props) {
        super(props);

        this.onToken = this.onToken.bind(this);
    }

    onToken(token) {
        fetch('/save-stripe-token', {
            method: 'POST',
            body: JSON.stringify(token),
        }).then(response => {
            response.json().then(data => {
                alert(`We are in business, ${data.email}`);
            });
        });
    }

    render() {
        return (
            <StripeCheckout
                name="We Do"
                description={this.props.description || 'Complete task'}
                panelLabel="Pay"
                amount={this.props.amount || 2000}
                currency="GBP"
                token={this.onToken}
                stripeKey="pk_test_TYooMQauvdEDq54NiTphI7jx"
            >
                <Button className="get-it-done-button">Get it done</Button>
            </StripeCheckout>
        )
    }
}