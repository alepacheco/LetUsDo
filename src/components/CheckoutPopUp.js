import React from 'react';
import Modal from 'react-bootstrap/Modal';
import Form from 'react-bootstrap/Form';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import CheckoutFormWrapper from './stripe/CheckoutFormWrapper';
import { bindActionCreators } from 'redux';

import '../styles/components/CheckoutPopUp.css';
import * as actions from '../actions/checkoutPopUpActions';

const validateEmail = email => {
  // eslint-disable-next-line no-useless-escape
  const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
  return re.test(String(email).toLowerCase());
}

class CheckoutPopUp extends React.Component {
  constructor(props) {
    super(props);

    this.onChangeEmail = this.onChangeEmail.bind(this);
  }

  onChangeEmail(event) {
    const emailValid = validateEmail(event.target.value);

    this.props.actions.setCheckoutEmailValid(emailValid);
    this.props.actions.setCheckoutEmail(event.target.value);
  }

  render() {
    return (
      <Modal
        show={this.props.show}
        onHide={this.props.onHide}
        size="lg"
        aria-labelledby="contained-modal-title-vcenter"
        centered
      >
        <Modal.Header closeButton>
          <Modal.Title id="contained-modal-title-vcenter">
            Confirm order
          </Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <div className="task-description-title">
            Task description
          </div>
          <div className="task-description-content">
            {this.props.taskText || 'No description specified'}
          </div>
          <Form>
            <Form.Group controlId="formBasicEmail" className="email-form">
              <Form.Control
                type="email"
                placeholder="Enter email"
                onChange={this.onChangeEmail}
                value={this.props.email} />
            </Form.Group>
          </Form>

          <CheckoutFormWrapper />
        </Modal.Body>

      </Modal>
    );
  }
}

const mapStateToProps = (state) => ({
  taskText: state.taskModal.taskText,
  email: state.checkoutPopUp.email
});

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(actions, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CheckoutPopUp);

CheckoutPopUp.propTypes = {
  onHide: PropTypes.func,
  taskText: PropTypes.string,
  email: PropTypes.string,
  show: PropTypes.bool,
  actions: PropTypes.shape({
    setCheckoutEmailValid: PropTypes.func,
    setCheckoutEmail: PropTypes.func
  }),
};