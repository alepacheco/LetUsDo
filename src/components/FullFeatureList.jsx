import React from 'react';
// import '../styles/components/FullFeature.scss';
import { FullFeature } from './FullFeature';
import Selection from '../static/images/undraw_selection.svg';
import Sprint from '../static/images/undraw_design_sprint.svg';
import Park from '../static/images/undraw_at_the_park.svg';

export const FullFeatureList = ({}) => {
  return (
    <div className="full-feature-list">
      <FullFeature Image={Sprint} />
      <FullFeature Image={Selection} imageSide="right" />
      <FullFeature Image={Park} />
    </div>
  );
};

export default FullFeatureList;
