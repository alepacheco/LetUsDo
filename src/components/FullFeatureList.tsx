import React from 'react';
import { FullFeature } from './FullFeature';
// @ts-ignore
import Selection from '../static/images/undraw_selection.svg';
// @ts-ignore
import Sprint from '../static/images/undraw_design_sprint.svg';
// @ts-ignore
import Park from '../static/images/undraw_at_the_park.svg';

export const FullFeatureList = () => (
  <div className="full-feature-list">
    <FullFeature Image={Sprint} title="Why we do it" imageSide="right" hideImage>
      We believe prioritisation is key if we want to achieve work-life balance, ||unfortunately life
      its messy and some things always come up when there is no one else to help out.|| With
      LetUsDo.app we want to change that. We literally give you an extra hand that is always there
      to help in whatever situation you need it.
    </FullFeature>

    <FullFeature Image={Selection} title="What do we do">
      We help people and organisations automate and delegate simple task in a on demand way. Using
      our website or out API we find and arrange someone to solve your task without any overhead on
      your side. We solve your problems and donâ€™t create more overhead in your busy life.
    </FullFeature>

    <FullFeature Image={Park} title="How do we do it" imageSide="right">
      When we receive a task from you we will assign it to one of our internal staff and then they
      will do whatever needs to be done to complete it\* They will only contact you in case you
      specify it or if in ultimate necessity. We want to give a complete smooth experience and
      complete the task without bothering you, any inconvenience that happens is up to our staff to
      resolve. \* Tasks needs to be simple enough to be completed in around an hour time
    </FullFeature>
  </div>
);

export default FullFeatureList;
