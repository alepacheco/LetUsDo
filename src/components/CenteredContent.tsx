import React from 'react';
import 'src/styles/components/CenteredContent.scss';

type CenteredContentProps = {
  reverse?: boolean;
};

export const CenteredContent: React.FC<CenteredContentProps> = ({ children, reverse }) => {
  return <div className={`centered-content ${reverse ? 'reverse' : ''}`}>{children}</div>;
};

export default CenteredContent;
