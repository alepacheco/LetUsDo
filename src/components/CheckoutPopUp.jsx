import React from 'react';
import Modal from 'react-bootstrap/Modal';
import Form from 'react-bootstrap/Form';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import CheckoutFormWrapper from './stripe/CheckoutFormWrapper';
import '../styles/components/CheckoutPopUp.css';
import * as checkoutPopUpActions from '../actions/checkoutPopUpActions';
import * as taskModalActions from '../actions/taskModalActions';

const validateEmail = email => {
  // eslint-disable-next-line no-useless-escape
  const validEmailRegex = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
  return validEmailRegex.test(String(email).toLowerCase());
};

export class CheckoutPopUp extends React.Component {
  constructor(props) {
    super(props);

    this.onChangeEmail = this.onChangeEmail.bind(this);
    this.onHideCheckoutPopUp = this.onHideCheckoutPopUp.bind(this);
  }

  onChangeEmail(event) {
    const emailValid = validateEmail(event.target.value);

    this.props.actions.setCheckoutEmailValid(emailValid);
    this.props.actions.setCheckoutEmail(event.target.value);
  }

  onHideCheckoutPopUp() {
    this.props.actions.setDialog('closed');
  }

  render() {
    return (
      <Modal
        show={this.props.checkoutPopupState !== 'closed'}
        onHide={this.onHideCheckoutPopUp}
        size="lg"
        aria-labelledby="contained-modal-title-vcenter"
        centered
      >
        <Modal.Header closeButton>
          <Modal.Title id="contained-modal-title-vcenter">Confirm order</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <div className="task-description-title">Task description</div>
          <div className="task-description-content">
            {this.props.taskText || 'No description specified'}
          </div>
          <Form>
            <Form.Group controlId="formBasicEmail" className="email-form">
              <Form.Control
                type="email"
                placeholder="Enter email"
                onChange={this.onChangeEmail}
                value={this.props.email}
              />
            </Form.Group>
          </Form>

          <CheckoutFormWrapper />
        </Modal.Body>
      </Modal>
    );
  }
}

const mapStateToProps = state => ({
  taskText: state.taskModal.taskText,
  email: state.checkoutPopUp.email,
  purchaseCompleted: state.checkoutPopUp.purchaseCompleted,
  checkoutPopupState: state.taskModal.checkoutPopupState
});

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(
      {
        ...taskModalActions,
        ...checkoutPopUpActions
      },
      dispatch
    )
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CheckoutPopUp);

CheckoutPopUp.propTypes = {
  taskText: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
  checkoutPopupState: PropTypes.string.isRequired,
  actions: PropTypes.shape({
    setCheckoutEmailValid: PropTypes.func,
    setCheckoutEmail: PropTypes.func,
    setDialog: PropTypes.func
  }).isRequired
};
