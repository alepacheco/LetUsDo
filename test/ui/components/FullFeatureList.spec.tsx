import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { FullFeatureList } from 'components/FullFeatureList';

describe('<FullFeatureList />', () => {
  it('renders', () => {
    const wrapper = shallow(<FullFeatureList />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
