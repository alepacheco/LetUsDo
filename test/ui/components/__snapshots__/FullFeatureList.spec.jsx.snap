// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FullFeatureList /> renders 1`] = `
<div
  className="full-feature-list"
>
  <FullFeature
    Image={
      Object {
        "process": [Function],
      }
    }
    imageSide="left"
    title="What can we do?"
  >
    We can do both 
    <b>
      online or in person tasks in London
    </b>
    .
    <br />
    Anything that you can delegate to somebody without much domain specific knowledge that can be done in 
    <b>
      between an hour
    </b>
     is a good task for us.
    <br />
    We will try to do whatever you need, if we can’t make it you’ll receive a full refund
  </FullFeature>
  <FullFeature
    Image={
      Object {
        "process": [Function],
      }
    }
    imageSide="right"
    title="How does it work?"
  >
    Once you find a task that you would like to delegate to us, write all the details down that we will need to perform it.
    <br />
    Include things like 
    <b>
      timelines
    </b>
     if any and other details.
    <br />
    Once you have the description ready you can summit it to us, we will receive it and just do it. If we need more information to complete it
    <br />
    <b>
      we will contact you to the email summited
    </b>
     or elsewhere provided. If we find that the
     
    <b>
      task is ineligible
    </b>
     by been too complex or it cannot be competed in an hour we will give you a 
    <b>
      full refund.
    </b>
  </FullFeature>
  <FullFeature
    Image={
      Object {
        "process": [Function],
      }
    }
    imageSide="left"
    title="Need help?"
  >
    If you have any issue with your order contact us by email at:
     
    <a
      href="mailto:support@letusdo.app"
    >
      support@letusdo.app
    </a>
  </FullFeature>
</div>
`;
